cnf-certification-test/platform/cnffsdiff/fsdiff.go

```diff
@@ -53,32 +73,52 @@ func NewFsDiffTester(client clientsholder.Command) *FsDiff {
 	}
 }
 
+//nolint:funlen
 func (f *FsDiff) RunTest(ctx clientsholder.Context, containerUID string) {
-	expected := []string{varlibrpm, varlibdpkg, bin, sbin, lib, usrbin, usrsbin, usrlib}
 	output, outerr, err := f.ClientHolder.ExecCommandContainer(ctx, fmt.Sprintf("chroot /host podman diff --format json %s", containerUID))
 	if err != nil {
-		logrus.Errorln("can't execute command on container ", err)
+		f.Error = fmt.Errorf("can't execute command on container: %w", err)
 		f.result = testhelper.ERROR
 		return
 	}
 	if outerr != "" {
+		f.Error = fmt.Errorf("stderr log received when running fsdiff test: %s", outerr)
 		f.result = testhelper.ERROR
-		logrus.Errorln("error when running fsdiff test ", outerr)
 		return
 	}
 
-	for _, exp := range expected {
-		// panic if the expression is wrong
-		r := regexp.MustCompile(exp)
-		if r.MatchString(output) {
-			logrus.Error("an installed package found on ", exp)
-			f.result = testhelper.FAILURE
-			return
-		}
-	}
 	// see if there's a match in the output
 	logrus.Traceln("the output is ", output)
-	f.result = testhelper.SUCCESS
+
+	diff := fsDiffJSON{}
+	err = json.Unmarshal([]byte(output), &diff)
+	if err != nil {
+		f.Error = fmt.Errorf("failed to unmarshall podman diff's json output: %s, err: %w", output, err)
+		f.result = testhelper.ERROR
+		return
+	}
+
+	// Check for deleted folders.
+	for _, deletedFolder := range diff.Deleted {
+		if _, exist := targetFolders[deletedFolder]; exist {
+			logrus.Tracef("Container's folder %s has been deleted.", deletedFolder)
+			f.DeletedFolders = append(f.DeletedFolders, deletedFolder)
+		}
+	}
+
+	// Check for changed folders.
+	for _, changedFolder := range diff.Changed {
```

Comment:
you should check if any of the target folders is a prefix of the changed folder.
All our target folder were terminated by * before

Reply:
I mean what if a user change something under /bin/go
then the modified folder would be '/bin/go' which is not in our list.